internal/controller/pipelinegate_controller.go

```diff
@@ -53,59 +56,105 @@ type PipelineGateReconciler struct {
 //+kubebuilder:rbac:groups=deployments.plural.sh,resources=pipelinegates/finalizers,verbs=update
 //+kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;watch;create;update;patch;delete;deletecollection
 
-func (r *PipelineGateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *PipelineGateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
 	log := log.FromContext(ctx).WithValues("PipelineGate", req.NamespacedName)
 
-	gate := &v1alpha1.PipelineGate{}
-
-	// get pipelinegate
-	if err := r.Get(ctx, req.NamespacedName, gate); err != nil {
+	crGate := &v1alpha1.PipelineGate{}
+	if err := r.Get(ctx, req.NamespacedName, crGate); err != nil {
 		if apierrs.IsNotFound(err) {
-			log.V(1).Info("PipelineGate CR not found - skipping.", "Namespace", gate.Namespace, "Name", gate.Name)
+			log.V(1).Info("PipelineGate CR not found - skipping.", "Namespace", crGate.Namespace, "Name", crGate.Name)
 			return ctrl.Result{}, nil
 		}
 		log.Error(err, "Unable to fetch PipelineGate.")
 		return ctrl.Result{}, err
 	}
 
+	cachedGate, err := r.GateCache.Get(crGate.Spec.ID)
+	if err != nil {
+		log.Error(err, "Unable to fetch PipelineGate from cache, this gate probably doesn't exist in the console.")
+		return ctrl.Result{}, err
+	}
+
+	scope, _ := NewPipelineGateScope(ctx, r.Client, crGate)
+	defer func() {
+		attrs, _ := crGate.Status.GateUpdateAttributes()
+		sha, _ := utils.HashObject(attrs)
```

Comment:
I think it may be risky to skip these errors (even if the chance of error is low). Setting wrong, i.e. empty, SHA can cause issues later.